pkg/cloudprovider/providers/aws/volumes.go

```diff
@@ -141,7 +141,10 @@ func (c *Cloud) checkIfAttachedToNode(diskName KubernetesVolumeID, nodeName type
 		}
 
 		awsDiskInfo.ec2Instance = instanceInfo
-		awsDiskInfo.nodeName = mapInstanceToNodeName(instanceInfo)
+		awsDiskInfo.nodeName, err = mapInstanceToNodeName(instanceInfo, c.cfg.Global.ExplicitNodeNames)
```

Comment:
This is going to be a problem, if someone switches naming topology midway through, then disks could potentially get orphaned.